### PR TITLE
EDD-25 verifying the host for downloads

### PR DESCRIPTION
**Pull request by jaybarra**
Adds validation to downloadLink sources. Compares them against a list of known and trusted sources before allowing downloads to continue.

### CODE REVIEW

File: docs/GET_LINKS.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ After the first page of links is received EDD will start the download. This will
 
 ## Expected Response
 
-It is very imported that your API returns the correct response to EDD
+It is very imported that your API returns the correct response to EDD. The response schema is located at [getLinksSchema.json](../src/main/getLinksSchema.json).
```
Comment (eudoroolivares2016) [inline docs/GET_LINKS.md:15]:
```suggestion
It is very important that your API returns the correct response to EDD. The response schema is located at [getLinksSchema.json](../src/main/getLinksSchema.json).
```

File: docs/USE_EDD.md
Comment (macrouch) [inline docs/USE_EDD.md:1]: At the top of this file, under the bit about being designed to reused, add some details about adding your application's API endpoints to the trustedSources.json file

File: docs/GET_LINKS.md
Comment (macrouch) [inline docs/GET_LINKS.md:1]: The `getLinks` link was meant to link to the USE_EDD.md file, can you update that one too

File: src/main/utils/fetchLinks.ts [hunk 2]
```diff
@@ -26,6 +33,25 @@ import downloadStates from '../../app/constants/downloadStates'
...
+
+  const host = link
+    .replace(/^https?:\/\//i, '')
+    .split('/')
+    .at(0)
+    ?.toLowerCase()
+    ?.split(':')
+    ?.at(0)
+  console.debug(`Checking [${host}] for matching trusted host`)
+  return !!trustedSources[host]
```
Comment (DanielJDufour) [inline src/main/utils/fetchLinks.ts:52]: minor point of personal preference, but could also do
```js
return host in trustedSources
```

**Review by DanielJDufour — APPROVED** [review]

**Review by dpesall — APPROVED** [review]

**Pull request merged**